src/Storages/MergeTree/ParallelReplicasReadingCoordinator.h

```diff
@@ -30,11 +30,15 @@ class ParallelReplicasReadingCoordinator
     /// needed to report total rows to read
     void setProgressCallback(ProgressCallback callback);
 
+    /// Participating replicas count may be less than replicas count in a shard.
+    void adjustParticipatingReplicasCount(size_t count) { participating_replicas_count = count; }
+
 private:
     void initialize(CoordinationMode mode);
 
     std::mutex mutex;
     const size_t replicas_count{0};
+    size_t participating_replicas_count{0};
```

Comment:
There is no need to have this member. We can just create a coordinator when the number of replicas to use for the query is determined.